### DISCUSSION TITLE
CMake on Windows: config not working

### DISCUSSION THREAD

**Discussion by nanocom** [ccache/ccache#1436] [Support]
### Description ###

The doc suggests that we put this line in cmakelists.txt here https://github.com/ccache/ccache/wiki/MS-Visual-Studio
`set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT "$<$<CONFIG:Debug,RelWithDebInfo>:Embedded>")`
But it doesn't work since the flag /Zi is not replaced by /Z7
So we had to manually add these lines, and it worked:
```
string(REGEX REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
string(REGEX REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
```

**Comment by cristianadam** [comment]
Which CMake version are you using? Which version have you set in `cmake_minimum_required` ?

[CMAKE_MSVC_DEBUG_INFORMATION_FORMAT](https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_DEBUG_INFORMATION_FORMAT.html) has been added in CMake version 3.25.

See the note from the above documentation:

> Note This variable has effect only when policy [CMP0141](https://cmake.org/cmake/help/latest/policy/CMP0141.html#policy:CMP0141) is set to NEW prior to the first [project()](https://cmake.org/cmake/help/latest/command/project.html#command:project) or [enable_language()](https://cmake.org/cmake/help/latest/command/enable_language.html#command:enable_language) command that enables a language using a compiler targeting the MSVC ABI.

**Comment by nanocom** [comment]
i'm using cmake 3.29, i'll try to put the policy before project() thx
